### PR TITLE
Testing graphql bagpart.

### PR DESCRIPTION
**Pull request by jtkech**
@jrestall 

Just to show what i quicly did on my side to make bagpart working. Tried in the context of the landing page of the agency setup. Also tried after adding a new bagpart through the admin ui.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs [hunk 1]
```diff
@@ -23,11 +23,10 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
 
             foreach (var part in contentTypeDefinition.Parts)
             {
-                // Check if another builder has already added a field for this part.
-                if (contentItemType.HasField(part.Name)) continue;
+                var partName = part.Name;
 
-                // This builder only handles parts with fields.
-                if (!part.PartDefinition.Fields.Any()) continue;
```
Comment (jrestall) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs]: I don't yet understand why this check needs to be removed. The code in this class only operates on fields, so if the part has no fields we probably shouldn't create an empty field?

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs [hunk 2]
```diff
@@ -23,11 +23,10 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
...
+                // Check if another builder has already added a field for this part.
+                if (contentItemType.HasField(partName)) continue;
                 
                 // When the part has the same name as the content type, it is the main part for
                 // the content type's fields so we collpase them into the parent type.
```
Comment (jrestall) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs]:
```suggestion
                // the content type's fields so we collapse them into the parent type.
```

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/TypedContentTypeBuilder.cs [hunk 3]
```diff
@@ -42,7 +42,7 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
                         partName,
                         resolve: context =>
                         {
-                            var nameToResolve = context.ReturnType.Name;
+                            var nameToResolve = partName;
```
Comment (jrestall) [inline src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/TypedContentTypeBuilder.cs]: I like it!

File: src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/DynamicContentTypeBuilder.cs]: Yes, it still works if i put it back, maybe i needed it at some point. i will revert this change.

**Comment by jtkech** [comment]
Not yet tried to add a non named bagpart.

Otherwise, from the landing page the following works. So, seems that it fixes #2674.

graphql2

**Pull request merged**